File: cdk_integration_tests/src/typescript/SecretManagerSecretEncrypted/fail__2__.ts
import { App, Stack } from 'aws-cdk-lib';
import * as secretsmanager from 'aws-cdk-lib/aws-secretsmanager';
import * as kms from 'aws-cdk-lib/aws-kms';
import { Construct } from 'constructs';

class MySecretsStack extends Stack {
  constructor(scope: Construct, id: string, props?: {}) {
    super(scope, id, props);

    // Define a SecretsManager secret with KMS key ID
    const mySecret = new secretsmanager.Secret(this, 'MySecret', {
      secretName: 'MySecretName',
      encryptionKey: kms.Key.fromKeyArn(this, 'MyKmsKey', 'arn:aws:kms:REGION:ACCOUNT_ID:key/aws/KMS_KEY_ID'),
    });
  }
}

class MySecretsStack2 extends Stack {
  constructor(scope: Construct, id: string, props?: {}) {
    super(scope, id, props);

    // Define a SecretsManager secret without specifying KMS key ID
    const mySecret = new secretsmanager.Secret(this, 'MySecret', {
      secretName: 'MySecretName',
    });
  }
}

const app = new App();
new MySecretsStack(app, "MySecretsStack");
new MySecretsStack2(app, "MySecretsStack2");
app.synth();

File: cdk_integration_tests/src/python/SecretManagerSecretEncrypted/fail__2__.py
from aws_cdk import core
from aws_cdk import aws_secretsmanager as secretsmanager

class MySecretsStack(core.Stack):
    def __init__(self, scope: core.Construct, id: str, **kwargs) -> None:
        super().__init__(scope, id, **kwargs)

        # Define a SecretsManager secret with KMS key ID containing "aws/"
        my_secret = secretsmanager.Secret(
            self, 'MySecret',
            secret_name='MySecretName',
            kms_key_id='arn:aws:kms:REGION:ACCOUNT_ID:key/aws/KMS_KEY_ID'
        )

class MySecretsStack2(core.Stack):
    def __init__(self, scope: core.Construct, id: str, **kwargs) -> None:
        super().__init__(scope, id, **kwargs)

        my_secret = secretsmanager.Secret(
            self, 'MySecret',
            secret_name='MySecretName',
        )

app = core.App()
MySecretsStack(app, "MySecretsStack")
app.synth()
